UnityProject/Assets/Scripts/Editor/Tools/AddressableToGameObjectTree.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.AddressableAssets;
+
+namespace Assets.Scripts.Editor.Tools
+{
+	/// <summary>
+	/// Will convert all addressables find in the assets tree to a game object that respect the same hierarchy.
+	/// All folders will be converted in a GameObject that contains either child GameObject (folders)
+	/// or have a AssetReferenceLibrary that will contains AssetReferences
+	/// All prefab addressables will be put in the AssetReferenceLibrary of the right GameObject in the hierarchy.
+	/// </summary>
+	public class AddressableToGameObjectTree : EditorWindow
+	{
+		string path;
+		GameObject rootObject;
+		private string[] assetsPaths;
+
+		[MenuItem("Tools/Addressables to GameObject Tree Converter")]
+		private static void Init()
+		{
+			// Get existing open window or if none, make a new one:
+			AddressableToGameObjectTree window = (AddressableToGameObjectTree)GetWindow(typeof(AddressableToGameObjectTree));
+			window.titleContent.text = "Addressables to GameObject Tree Converter";
+			window.Show();
+		}
+
+		private void SelectPath()
+		{
+			string baseSTR = EditorUtility.OpenFolderPanel("Select folder to save prefabs", Application.dataPath, "prefabs");
+			if (baseSTR != "")
+			{
+				if (-1 == baseSTR.IndexOf(Application.dataPath, System.StringComparison.InvariantCultureIgnoreCase))
+				{
+					EditorUtility.DisplayDialog("Assets folder error", "Target folder must be any child directory to : \"" + Application.dataPath + "\"", "OK");
+					path = "";
+				}
+				else
+				{
+					path = baseSTR.Replace(Application.dataPath, "Assets");
+				}
+			}
+		}
+
+		private void ConvertToGameObjectRecursive(GameObject parentObject, string path)
+		{
+			string objectName = path.Split('/').Last();
+
+			GameObject gameObject = new GameObject(objectName);
+			gameObject.transform.parent = parentObject.transform;
+			Debug.Log($"Adding gameobject {objectName}");
+
+			// List of all prefabs exactly at path (without their file name)
+			List<string> prefabsAtPath = assetsPaths.Where(p => string.Join("/", p.Split('/').Reverse().Skip(1).Reverse()).Equals(path) && p.Contains(".prefab")).ToList();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'prefabsAtPath'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)